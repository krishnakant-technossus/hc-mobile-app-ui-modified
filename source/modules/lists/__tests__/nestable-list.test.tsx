// npx jest ./source/modules/lists/__tests__/nestable-list.test.ts

import * as React from 'react';
import {fireEvent, render} from '@testing-library/react-native';

import {
  NestableList,
  NestableListParentItem,
  NestableListChildItem,
} from '../nestable-list';
import ThemeProvider from '../../theming/theme-provider';
import defaultTheme from '../../theming/default-theme';

const onPressParentMock = jest.fn();
const onPressChildMock = jest.fn();

const nestableListItems = [
  {
    label: 'Campus',
    onPress: onPressParentMock,
    id: 'app-list-item-campus',
    children: [
      {
        label: 'Lake Forest',
        id: 'app-list-item-child-lake-forest',
        onPress: onPressChildMock,
      },
      {
        label: 'Laguna Woods',
        id: 'app-list-item-child-laguna-woods',
        onPress: onPressChildMock,
      },
      {
        label: 'San Diego',
        id: 'app-list-item-child-san-diego',
        onPress: onPressChildMock,
      },
    ],
  },
  {
    label: 'Category',
    onPress: onPressParentMock,
    id: 'app-list-item-category',
  },
  {
    label: 'Tags',
    onPress: onPressParentMock,
    id: 'app-list-item-tags',
  },
];

describe('NestableList', () => {
  describe('Rendering', () => {
    it('renders itself correctly', () => {
      const wrapper = render(
        <ThemeProvider theme={defaultTheme}>
          <NestableList items={nestableListItems} />
        </ThemeProvider>,
      );

      const list = wrapper.getByTestId('list');

      expect(list).toBeDefined();

      expect(wrapper).toMatchSnapshot();
    });

    it('renders the parent items correctly', () => {
      const wrapper = render(
        <ThemeProvider theme={defaultTheme}>
          <NestableList items={nestableListItems} />
        </ThemeProvider>,
      );

      const parentItems = wrapper.getAllByTestId('parent');

      expect(parentItems).toHaveLength(nestableListItems.length);

      const labels = parentItems.map(item => {
        const label = item.findByProps({testID: 'parent-label'});

        return label.props.children;
      });

      labels.forEach((label, ndx) => {
        expect(label).toEqual(nestableListItems[ndx].label);
      });
    });
  });
});

describe('NestableListParentItem', () => {
  const [item] = nestableListItems;

  describe('Rendering', () => {
    it('renders itself correctly', () => {
      const wrapper = render(
        <ThemeProvider theme={defaultTheme}>
          <NestableListParentItem {...item} />
        </ThemeProvider>,
      );

      const parentItem = wrapper.getByTestId('parent');

      expect(parentItem).toBeDefined();

      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('Interactions', () => {
    it('should call onPress method when pressed', () => {
      const wrapper = render(
        <ThemeProvider theme={defaultTheme}>
          <NestableListParentItem {...item} />
        </ThemeProvider>,
      );

      const parentItem = wrapper.getByTestId('parent');

      expect(parentItem).toBeDefined();

      fireEvent.press(parentItem);

      expect(onPressParentMock).toHaveBeenCalledTimes(1);
    });
  });
});

describe('NestableListChildItem', () => {
  const [item] = nestableListItems;
  const [child] = item.children;

  describe('Rendering', () => {
    it('renders itself correctly', () => {
      const wrapper = render(
        <ThemeProvider theme={defaultTheme}>
          <NestableListChildItem {...child} />
        </ThemeProvider>,
      );

      const childItem = wrapper.getByTestId('child');

      expect(childItem).toBeDefined();

      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('Interactions', () => {
    it('should call onPress method when pressed', () => {
      const wrapper = render(
        <ThemeProvider theme={defaultTheme}>
          <NestableListChildItem {...child} />
        </ThemeProvider>,
      );

      const childItem = wrapper.getByTestId('child');

      expect(childItem).toBeDefined();

      fireEvent.press(childItem);

      expect(onPressChildMock).toHaveBeenCalledTimes(1);
    });
  });
});
