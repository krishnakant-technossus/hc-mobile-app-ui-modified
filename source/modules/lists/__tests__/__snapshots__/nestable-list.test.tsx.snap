// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NestableList Rendering renders itself correctly 1`] = `
<View
  style={
    Array [
      Object {},
    ]
  }
  testID="list"
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#DBE0E3",
          "height": 1,
          "width": "100%",
        },
      ]
    }
    testID="divider"
  />
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    hitSlop={
      Object {
        "bottom": 12,
        "left": 12,
        "right": 12,
        "top": 12,
      }
    }
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "opacity": 1,
        "paddingBottom": 12.5,
        "paddingHorizontal": 11,
        "paddingTop": 13.75,
      }
    }
    testID="parent"
  >
    <Text
      style={
        Array [
          Object {
            "color": "#1C2530",
            "fontFamily": "System",
            "fontSize": 14,
            "fontWeight": "500",
            "textAlign": "left",
          },
        ]
      }
      testID="parent-label"
    >
      Campus
    </Text>
    <Image
      source={
        Object {
          "testUri": "../../../source/images/chevron-right-black.png",
        }
      }
      style={
        Array [
          Object {
            "height": 12,
            "width": 12,
          },
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#DBE0E3",
          "height": 1,
          "width": "100%",
        },
      ]
    }
    testID="divider"
  />
  <View
    style={
      Array [
        Object {
          "paddingBottom": 22,
          "paddingLeft": 22,
          "paddingRight": 22,
          "paddingTop": 22,
        },
      ]
    }
    testID="children"
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      hitSlop={
        Object {
          "bottom": 12,
          "left": 12,
          "right": 12,
          "top": 12,
        }
      }
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 13.75,
          "opacity": 1,
        }
      }
      testID="child"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#1C2530",
              "fontFamily": "System",
              "fontSize": 14,
              "fontWeight": "400",
              "textAlign": "left",
            },
          ]
        }
        testID="child-label"
      >
        Lake Forest
      </Text>
      <Image
        source={
          Object {
            "testUri": "../../../source/images/close-light-black.png",
          }
        }
        style={
          Array [
            Object {
              "height": 8,
              "width": 8,
            },
          ]
        }
      />
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      hitSlop={
        Object {
          "bottom": 12,
          "left": 12,
          "right": 12,
          "top": 12,
        }
      }
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 13.75,
          "opacity": 1,
        }
      }
      testID="child"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#1C2530",
              "fontFamily": "System",
              "fontSize": 14,
              "fontWeight": "400",
              "textAlign": "left",
            },
          ]
        }
        testID="child-label"
      >
        Laguna Woods
      </Text>
      <Image
        source={
          Object {
            "testUri": "../../../source/images/close-light-black.png",
          }
        }
        style={
          Array [
            Object {
              "height": 8,
              "width": 8,
            },
          ]
        }
      />
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      hitSlop={
        Object {
          "bottom": 12,
          "left": 12,
          "right": 12,
          "top": 12,
        }
      }
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 0,
          "opacity": 1,
        }
      }
      testID="child"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#1C2530",
              "fontFamily": "System",
              "fontSize": 14,
              "fontWeight": "400",
              "textAlign": "left",
            },
          ]
        }
        testID="child-label"
      >
        San Diego
      </Text>
      <Image
        source={
          Object {
            "testUri": "../../../source/images/close-light-black.png",
          }
        }
        style={
          Array [
            Object {
              "height": 8,
              "width": 8,
            },
          ]
        }
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#DBE0E3",
          "height": 1,
          "width": "100%",
        },
      ]
    }
    testID="divider"
  />
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    hitSlop={
      Object {
        "bottom": 12,
        "left": 12,
        "right": 12,
        "top": 12,
      }
    }
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "opacity": 1,
        "paddingBottom": 12.5,
        "paddingHorizontal": 11,
        "paddingTop": 13.75,
      }
    }
    testID="parent"
  >
    <Text
      style={
        Array [
          Object {
            "color": "#1C2530",
            "fontFamily": "System",
            "fontSize": 14,
            "fontWeight": "500",
            "textAlign": "left",
          },
        ]
      }
      testID="parent-label"
    >
      Category
    </Text>
    <Image
      source={
        Object {
          "testUri": "../../../source/images/chevron-right-black.png",
        }
      }
      style={
        Array [
          Object {
            "height": 12,
            "width": 12,
          },
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#DBE0E3",
          "height": 1,
          "width": "100%",
        },
      ]
    }
    testID="divider"
  />
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    hitSlop={
      Object {
        "bottom": 12,
        "left": 12,
        "right": 12,
        "top": 12,
      }
    }
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "opacity": 1,
        "paddingBottom": 12.5,
        "paddingHorizontal": 11,
        "paddingTop": 13.75,
      }
    }
    testID="parent"
  >
    <Text
      style={
        Array [
          Object {
            "color": "#1C2530",
            "fontFamily": "System",
            "fontSize": 14,
            "fontWeight": "500",
            "textAlign": "left",
          },
        ]
      }
      testID="parent-label"
    >
      Tags
    </Text>
    <Image
      source={
        Object {
          "testUri": "../../../source/images/chevron-right-black.png",
        }
      }
      style={
        Array [
          Object {
            "height": 12,
            "width": 12,
          },
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#DBE0E3",
          "height": 1,
          "width": "100%",
        },
      ]
    }
    testID="divider"
  />
</View>
`;

exports[`NestableListChildItem Rendering renders itself correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  hitSlop={
    Object {
      "bottom": 12,
      "left": 12,
      "right": 12,
      "top": 12,
    }
  }
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginBottom": 13.75,
      "opacity": 1,
    }
  }
  testID="child"
>
  <Text
    style={
      Array [
        Object {
          "color": "#1C2530",
          "fontFamily": "System",
          "fontSize": 14,
          "fontWeight": "400",
          "textAlign": "left",
        },
      ]
    }
    testID="child-label"
  >
    Lake Forest
  </Text>
  <Image
    source={
      Object {
        "testUri": "../../../source/images/close-light-black.png",
      }
    }
    style={
      Array [
        Object {
          "height": 8,
          "width": 8,
        },
      ]
    }
  />
</View>
`;

exports[`NestableListParentItem Rendering renders itself correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  hitSlop={
    Object {
      "bottom": 12,
      "left": 12,
      "right": 12,
      "top": 12,
    }
  }
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "opacity": 1,
      "paddingBottom": 12.5,
      "paddingHorizontal": 11,
      "paddingTop": 13.75,
    }
  }
  testID="parent"
>
  <Text
    style={
      Array [
        Object {
          "color": "#1C2530",
          "fontFamily": "System",
          "fontSize": 14,
          "fontWeight": "500",
          "textAlign": "left",
        },
      ]
    }
    testID="parent-label"
  >
    Campus
  </Text>
  <Image
    source={
      Object {
        "testUri": "../../../source/images/chevron-right-black.png",
      }
    }
    style={
      Array [
        Object {
          "height": 12,
          "width": 12,
        },
      ]
    }
  />
</View>
`;
